cli/pppp.py

```diff
@@ -13,10 +13,13 @@
 
 def pppp_open(env):
     with env.config.open() as cfg:
-        printer = cfg.printers[0]
+        if env.printer >= len(cfg.printers):
+            log.fatal(f"Printer number {env.printer} out of range, max printer number is {len(cfg.printers)-1} ")
+            return
+        printer = cfg.printers[env.printer]
 
         api = AnkerPPPPApi.open_lan(Duid.from_string(printer.p2p_duid), host=printer.ip_addr)
-        log.info("Trying connect over pppp")
+        log.info(f"Trying connect to printer {printer.p2p_duid} over pppp using ip {printer.ip_addr}")
```

Comment:
Excellent improvement - definitely much more informative :+1:

Reply:
I expanded upon this by adding in the printer's name to this output.